ykbh/src/lib.rs

```diff
@@ -5,36 +5,37 @@ use ykpack::{
     UnsignedInt,
 };
 use yktrace::sir::SIR;
-use std::mem::size_of;
-
-const ALIGN: usize = size_of::<usize>();
 
 pub struct SIRInterpreter {
     /// Pointer to allocated memory containing a frame's locals.
     locals: *mut u8,
     /// The offset of each Local into locals.
     offsets: Vec<usize>,
+    /// The layout of locals. Needed for deallocating locals upon drop.
+    layout: Layout,
 }
 
 impl SIRInterpreter {
     pub fn new(local_decls: &Vec<LocalDecl>) -> Self {
         // FIXME Soon this will be pre-computed and handed to us by SIR.
-        let mut size: usize = 0;
         let mut offsets = Vec::new();
+        let mut layout = Layout::from_size_align(0, 1).unwrap();
         for d in local_decls {
-            let padding = (ALIGN - (size % ALIGN)) % ALIGN;
-            size += padding;
-            offsets.push(size);
-            size += usize::try_from(SIR.ty(&d.ty).size()).unwrap();
+            let align = SIR.ty(&d.ty).align();
+            let size = SIR.ty(&d.ty).size();
+            let l = Layout::from_size_align(size.try_into().unwrap(), align.try_into().unwrap()).unwrap();
+            let (nl, s) = layout.extend(l).expect("Alignment error");
```

Comment:
I think `expect` is a) almost entirely pointless and b) clutters code. Let's use `unwrap`.

Reply:
Fair.